libraries/mbed/targets/hal/TARGET_Freescale/TARGET_K20XX/sleep.c

```diff
@@ -39,13 +39,27 @@ void deepsleep(void)
     SCB->SCR = 1<<SCB_SCR_SLEEPDEEP_Pos;
 
     __WFI();
-
     //Switch back to PLL as clock source if needed
     //The interrupt that woke up the device will run at reduced speed
     if (PLL_FLL_en) {
+        
+#if defined (TARGET_K20D50M)    
         if (MCG->C6 & (1<<MCG_C6_PLLS_SHIFT) != 0) /* If PLL */
             while((MCG->S & MCG_S_LOCK0_MASK) == 0x00U); /* Wait until locked */
         MCG->C1 &= ~MCG_C1_CLKS_MASK;
+#else    
+        MCG->C1 = (uint8_t)0x9Au;   
```

Comment:
Shouldn't we use macros available from mcu header files? Otherwise we need to convert hex to binary, check the reference manual and check what bits mean what... Similar to what you wrote in system init

```
from 
OSC0->CR = (uint8_t)0x0Au
to
OSC0->CR = OSC_CR_SC8P_MASK | OSC_CR_SC2P_MASK;
```